celery/worker/strategy.py

```diff
@@ -175,6 +165,18 @@ def task_message_handler(message, body, ack, reject, callbacks,
                 expires=req.expires and req.expires.isoformat(),
             )
 
+        if _does_info:
+            # Similar to `app.trace.info()`, we pass the formatting args as the
+            # `extra` kwarg for custom log handlers
+            context = {
+                'id': req.id,
+                'name': req.name,
+                'args': req.argsrepr,
+                'kwargs': req.kwargsrepr,
+                'eta': req.eta,
+            }
+            info(_app_trace.LOG_RECEIVED, context, extra={'data': context})
+
```

Comment:
For clarification, this very much does the same thing, however the order of which that happened just changed in this PR, hence logging is deferred to after sending events;

Former diff ensures, upon request creation, context is logged to celery before being processed.

Please, provide additional context to help clarify this PR:

- tests, ensuring the change effect is properly documented

Reply:
Effectively, you want to write a unit test to cover the flow change. At best, this is enough since this PR is not about implementation changes.

You should take a look at this file, [`test_strategy.py`](https://github.com/celery/celery/blob/main/t/unit/worker/test_strategy.py)

Reply:
good job @50-Course